repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/utilities/OpenStackIdentityV3API.scala

```diff
@@ -56,11 +55,17 @@ class OpenStackIdentityV3API(config: OpenstackIdentityV3Config, datastore: Datas
   private val identityServiceUri = config.getOpenstackIdentityService.getUri
   private val timeouts = Option(config.getCache).flatMap(cache => Option(cache.getTimeouts))
   private val cacheOffset =
-    timeouts.flatMap(timeouts => Option(timeouts.getVariance)).getOrElse(config.getCacheOffset)
+    timeouts.flatMap(timeouts => Option(timeouts.getVariance)).getOrElse(0)
+  // TODO: The Token and Group Cache TTL defaults (i.e. 600) will need moved to the XSD when the deprecated elements
+  // TODO: are removed and the attributes can directly have defaults.
   private val tokenCacheTtl =
-    timeouts.flatMap(timeouts => Option(timeouts.getToken).map(_.intValue())).getOrElse(config.getTokenCacheTimeout)
+  timeouts.flatMap(timeouts => Option(timeouts.getTokenElement).map(_.intValue()))
+    .getOrElse(timeouts.flatMap(timeouts => Option(timeouts.getToken).map(_.intValue()))
+      .getOrElse(600))
```

Comment:
Since the attributes are a new addition, it makes sense to me to give priority to the attributes over the elements (which are deprecated by this PR).

Reply:
Also, since `intValue()` does not mutate any state, I would recommend dropping the unnecessary parenthesis. So `.map(_.intValue)`.

Reply:
Also, I would do the same as before to simplify, and I would flatten out the structure. Like so:
```
timeouts.map(_.getToken).flatMap(Option.apply).map(_.intValue)
```

Reply:
Oh, and to further flatten out  the structure, I would make use of the `orElse` method.

Ultimately, I came up with:
```
timeouts.map(_.getToken).flatMap(Option.apply)
  .orElse(timeouts.map(_.getTokenElement).flatMap(Option.apply))
  .map(_.intValue)
  .getOrElse(600)
```

Or if you really wanted to minimize redundancy:
```
timeouts.flatMap(t => Option(t.getToken).orElse(Option(t.getTokenElement))
  .map(_.intValue)
  .getOrElse(600)
```

Reply:
Updated.